ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/ks/KsMaterialization.java

```diff
@@ -34,12 +35,15 @@
   private final Optional<WindowInfo> windowInfo;
   private final KsStateStore stateStore;
   private final Locator locator;
+  private final KafkaStreams kafkaStreams;
 
   KsMaterialization(
+      final KafkaStreams streams,
       final Optional<WindowInfo> windowInfo,
       final Locator locator,
       final KsStateStore stateStore
   ) {
+    this.kafkaStreams = requireNonNull(streams, "streams");
```

Comment:
Do we need to pass kafkastreams here? Why not use stateStore.getKafkaStreams()?

Reply:
I don't see the method `stateStore.getKafkaStreams()`? Am I missing something?

Reply:
Ah, I think you meant adding this method? Sure, we can do that

Reply:
My mistake, sorry, I didn't remember I added it in my PR so I assumed it was available. Yea adding it probably makes sense